app/services/organizations/importers/google_csv_import_service.rb

```diff
@@ -13,42 +13,62 @@ def initialize(file)
+            # Using Google Form headers
+            email = row[@email_header].downcase
+            csv_timestamp = Time.parse(row["Timestamp"])
 
-          if person.nil?
-            @no_match << [index, email]
-          elsif previously_matched_form_submission.present?
-            next
-          else
-            latest_form_submission = person.latest_form_submission
-            ActiveRecord::Base.transaction do
-              # This checks for the empty form submission that is added when a person is created
-              if latest_form_submission.csv_timestamp.nil? && latest_form_submission.form_answers.empty?
-                latest_form_submission.update!(csv_timestamp:)
-                create_form_answers(latest_form_submission, row)
-              else
-                # if the person submits a new/updated form,
-                # i.e. an additional row in the csv with the same email / different timestamp,
-                # a new form_submission will be created
-                create_form_answers(FormSubmission.create!(person:, csv_timestamp:), row)
+            person = Person.find_by(email:, organization: @organization)
+            previously_matched_form_submission = FormSubmission.where(person:, csv_timestamp:)
+
+            if person.nil?
+              @no_match << [index, email]
+            elsif previously_matched_form_submission.present?
+              next
+            else
+              latest_form_submission = person.latest_form_submission
+              ActiveRecord::Base.transaction do
+                # This checks for the empty form submission that is added when a person is created
+                if latest_form_submission.csv_timestamp.nil? && latest_form_submission.form_answers.empty?
+                  latest_form_submission.update!(csv_timestamp:)
+                  create_form_answers(latest_form_submission, row)
+                else
+                  # if the person submits a new/updated form,
+                  # i.e. an additional row in the csv with the same email / different timestamp,
+                  # a new form_submission will be created
+                  create_form_answers(FormSubmission.create!(person:, csv_timestamp:), row)
+                end
+                @count += 1
               end
-              @count += 1
             end
+          rescue => e
+            @errors << [index, e]
           end
-        rescue => e
-          @errors << [index, e]
         end
         Status.new(@errors.empty?, @count, @no_match, @errors)
       end
 
       private
 
+      def validate_file
+        raise FileTypeError unless @file.content_type == "text/csv"
+
+        first_row = CSV.foreach(@file.to_path).first
+        raise FileEmptyError if first_row.nil?
+
+        email_headers = ["Email", "email", "Email Address", "email address"]
+        email_headers.each do |e|
+          @email_header = e if first_row.include?(e)
+        end
```

Comment:
This is the simplest way I can think of at the moment to allow both "Email" and "email". Since I had to do this change anyways, it is easy enough to add new naming variation's so I added the stakeholders as well.

There isn't any extra logic to deal with the scenario of multiple headers having an email name.

Reply:
Yeah I like this.